Add PersistentStore type to store SubstrateIdentities in localStorage.

jnaviask opened the pull request:
## Description
Fixes #239. Adds a new type of store that writes to localStorage in addition to keeping a local list available. Also adds a clearing function run at page load, which checks timestamps and removes entries older than 10 minutes.

PR also converts the SubstrateIdentity controllers to utilize this cache, and ensures the changes are reflected in the view logic.

## Motivation and Context
Should speed up identity loading times from the chain, and remove flicker.

## How has this been tested?
Ran on Kusama and Edgeware and guaranteed identities load from chain and are stored correctly in PersistentStore. Set cleanup time to 10 seconds and ensured identities are removed and re-fetched. Ran on Edgeware Local and created and edited a new identity to ensure that feature worked as well.

## Have proper tags been added (for bug, enhancement, breaking change)?
- [x] yes

## Does this PR affect any server routes?
- [x] no

drewstone commented on client/scripts/controllers/chain/substrate/identities.ts:
```diff
@@ -100,20 +102,51 @@ class SubstrateIdentities implements StorageModule {
+    }
+    return identityExists$.pipe(
+      filter((v) => typeof v === 'boolean'),
+      takeWhile((v) => !v, true),
```
What is this `true` value doing, is that what is returned if !v?

jnaviask replied:
`true` here means if we see `v` then we return one final time (inclusive) and then close the observable, otherwise we keep returning `!v` whenever we see a new value. but we might be able to remove this entirely, based on how `identityExists$` returns. I'll look into it.

drewstone replied:
Should I hold off on merging then?

jnaviask replied:
i'll look into this tomorrow morning and give you the OK

jnaviask replied:
Reviewed it again -- it should be fine as it is. We don't want to accidentally trigger the deletion twice.

drewstone commented:
Seems to be a merge conflict to resolve as well @jnaviask

jnaviask commented:
> Seems to be a merge conflict to resolve as well @jnaviask

Will fix this.

jnaviask commented:
Regarding:
> Not sure why some proposalSlug functions are being removed

I removed `getProposalObservable` because it was no longer used (it used to help us figure out when to declare the app "loaded", but we've since built better functionality), and because removing it would allow me to remove the internal Observable member of our Store class, as it was the only call that used this feature of Stores (the first commit in this PR was Store cleanup).

The other function I removed, `proposalSlugsFromChain`, was only called by `getProposalObservable`, so I removed it for the sake of cleanup.
